### ISSUE TITLE
Proposal: New Plugin for Generic Helm Chart Deployment

### ISSUE THREAD

**Issue by prodanlabs** [devstream-io/devstream#424]
## Description

I personally think it is possible to create a generic helm plugin to deploy the application of helm charts
for example
```yaml
- name: helm-demo
  plugin: helm-charts-plugin
  options:
    #  helm charts address
    repo: YOUR_HELM_CHARTS
    namespace: K8S_NAMESPACE
    # helm charts deployment parameters
    helm_options: "parameter1","parameter2"
```
I don't know if anyone is interested in this feature :）

**Comment by daniel-hutao** [comment]
# Summary

Thanks for your proposal, it's a great idea. I'd like to give a design here.

## Scope

As you all know, we already have a lot of plugins implemented through the helm SDK. Maybe you have a question here: Why don't you implement all `helm-type plugin` deployments through a plugin names `helm-generic` plugin? Let me tell you.

If you simply deploy a tool through `helm install` command, you don't even need `dtm`, you just use `helm cli` directly. So what is the purpose of implementing a lot of helm-style plugins?

Maybe you already get it, we definitely need to do some extra work for it. For example, a tool may need to do Pre-configuration with storage before passing `helm install`. Or some cleanup needs to be done after `helm uninstall`. Or some additional robustness processing is required during the install process.

In conclusion, we believe that the deployment of most `helm charts` is not a simple one-line command, and the additional operations required for these deployment processes are also different. So we have developed separate plugins for the deployment of each helm-style tools.

**Comment by prodanlabs** [comment]
Sorry, I don't have time right now. If it affects the version plan may need to be assigned to someone else.
I might have time next month.

**Comment by prodanlabs** [comment]
Hi @daniel-hutao ，I was thinking, it seems that the `kubeprometheus` plugin can also be turned into a generic helm plugin.
